tests/covalent_dispatcher_tests/_core/execution_test.py

```diff
@@ -37,6 +39,28 @@
 
 TEST_RESULTS_DIR = "/tmp/results"
 
+DEFAULT_THREADS_PER_WORKERS = 1
+DEFAULT_N_WORKERS = dask.system.CPU_COUNT
+DEFAULT_MEM_PER_WORKER = "auto"
+
+
+@pytest.fixture
+def test_cluster():
+    cluster = LocalCluster(
+        n_workers=DEFAULT_N_WORKERS,
+        threads_per_worker=DEFAULT_THREADS_PER_WORKERS,
+        **{"memory_limit": DEFAULT_MEM_PER_WORKER},
+    )
+    yield cluster
+    cluster.close()
```

Comment:
Why don't we just return the cluster here instead of yielding? We can close it later. In fact, I think we can get away with just using a local executor here.